src/applications/verify/components/Verify.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import {
+  isAuthenticatedWithOAuth,
+  signInServiceName,
+} from 'platform/user/authentication/selectors';
+import {
+  VerifyIdmeButton,
+  VerifyLogingovButton,
+  verifyHandler,
+} from 'platform/user/authentication/components/VerifyButton';
+import { SERVICE_PROVIDERS } from 'platform/user/authentication/constants';
+
+const Verify = () => {
+  const isAuthenticated = useSelector(isAuthenticatedWithOAuth);
+  const loginServiceName = useSelector(signInServiceName); // Get the current SIS (e.g., idme or logingov)
+
+  useEffect(
+    () => {
+      // Automatically start verification if returning from login
+      const queryParams = new URLSearchParams(window.location.search);
+      if (queryParams.get('postLogin') === 'true') {
+        const policy =
+          loginServiceName === 'idme'
+            ? SERVICE_PROVIDERS.idme.policy
+            : SERVICE_PROVIDERS.logingov.policy;
+
+        verifyHandler({ policy, queryParams: {} });
+      }
+    },
+    [loginServiceName],
+  );
+
+  let buttonContent;
+
+  if (isAuthenticated) {
+    // Use the loginServiceName to determine which button to show
+    if (loginServiceName === 'idme') {
+      buttonContent = (
+        <VerifyIdmeButton
+          queryParams={{ operation: 'authenticated_verify_page' }}
+        />
+      );
+    } else if (loginServiceName === 'logingov') {
+      buttonContent = (
+        <VerifyLogingovButton
+          queryParams={{ operation: 'authenticated_verify_page' }}
+        />
+      );
+    }
+  } else {
+    buttonContent = (
+      <>
+        <VerifyLogingovButton
+          queryParams={{ operation: 'unauthenticated_verify_page' }}
+        />
+        <VerifyIdmeButton
+          queryParams={{ operation: 'unauthenticated_verify_page' }}
+        />
+      </>
+    );
+  }
```

Comment:
- Again you are 98% there which is great. The other `else` statement should be all unauthenticated in which case we want to force OAuth to be used as we won't have the `isAuthenticatedWithOAuth` because the user isn't signed in here.

```javascript
<>
        <VerifyLogingovButton
         forceOAuth
          queryParams={{ operation: 'unauthenticated_verify_page' }}
        />
        <VerifyIdmeButton
          forceOAuth
          queryParams={{ operation: 'unauthenticated_verify_page' }}
        />
      </>
```